src/fhirtypes/ElementDefinition.ts

```diff
@@ -1897,7 +1897,7 @@ export class ElementDefinition {
       (type === 'integer64' && /^[-]?\d+$/.test(value))
     ) {
       this.assignFHIRValue(`"${value}"`, value, exactly, type);
-    } else if (type == 'xhtml' && this.checkXhtml(value)) {
+    } else if (this.checkXhtml(value) && type == 'xhtml') {
```

Comment:
I think it would be better to check the value of `type` before calling `this.checkXhtml`. If you call `this.checkXhtml` first, you could get a warning from it even on an element of a different type.

Reply:
Done.